### ISSUE TITLE
Document gradient clipping with manual optimization

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#14598]
## 📚 Documentation

On [this page](https://pytorch-lightning.readthedocs.io/en/stable/model/manual_optimization.html?highlight=manual%20optimiztion), we should probably document the clip_gradients method that you can use in manual optimization, since it's not handled through the Trainer arguments. 

The docstring for this method should also be improved to mention manual optimization:

https://github.com/Lightning-AI/lightning/blob/95374440ce9b4f499f83b0e6b8cee0bc715aa18b/src/pytorch_lightning/core/module.py#L1474-L1491

In conjunction, also clarify when to use `configure_gradient_clipping` and when not (manual vs automatic opt). 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @borda @rohitgr7 @Felonious-Spellfire @akihironitta @krshrimali

**Comment by shenoynikhil** [comment]
Hey @awaelchli, I'd like to take this up!

**Comment by awaelchli** [comment]
@shenoynikhil Sure, go for it <3
